feat: restructure and improve CICD docs and examples (#135)
terramate-io/terramate-docs: cli/automation/github-actions/deployment-workflow.md

The following workflows are blueprints and need some adjustments to work for you.

Search for `CHANGEME` to adjust needed credentials details for AWS, Google Cloud, and Azure examples.

## Terramate Cloud support

When synchronizing deployments to Terramate Cloud, it is recommended to run a drift check right after the deployment.

> This drift check will be used to judge the health of the deployment. Even if the deployment succeeded, if can show a drift right away.

Reviewer: ```suggestion
This drift check will be used to validate the integrity of the deployment. Consider the following two scenarios:
- If a deployment succeeds, an immediate drift check will help detect if resources drift right away.
- If a deployment fails, an immediate drift will help to detect and understand partially applied plans.
```

      - name: Run OpenTofu init on changed stacks
        if: steps.list.outputs.stdout
        id: init
        run: |
          terramate run \
            --parallel 1 \
            --changed \
            -- \
            tofu init

      - name: Create OpenTofu plan on changed stacks
        if: steps.list.outputs.stdout
        id: plan
        run: |
          terramate run \
            --parallel 5 \
            --changed \
            -- \
            tofu plan -lock-timeout=5m -out out.otplan

      - name: Apply planned changes on changed stacks
        if: steps.list.outputs.stdout
        id: apply
        run: |
          terramate run \
            --parallel 5 \
            --changed \
            --sync-deployment \
            --tofu-plan-file=out.otplan \
            -- \
            tofu apply -input=false -auto-approve -lock-timeout=5m out.otplan
        env:
          GITHUB_TOKEN: ${{ github.token }}

      - name: Run drift detection
        if: steps.list.outputs.stdout && ! cancelled() && steps.apply.outcome != 'skipped'
        id: drift
        run: |
          terramate run \
            --parallel 5 \

>             --changed \

Reviewer: ```suggestion
            --changed \
            --continue-on-error \
```

Reviewer: Continue on error is missing here to finish the drift run in all stacks even if errors occur

            --sync-drift-status \
            --tofu-plan-file=drift.otplan \
            -- \
            tofu plan -out drift.otplan -detailed-exitcode
        env:

>           GITHUB_TOKEN: ${{ github.token }}

Reviewer: Maybe it makes sense to add an optional and commented reconcile step here that per default only runs on stacks that filter for `--tags reconcile`? For reference please see https://github.com/terramate-io/terramate-quickstart-aws/blob/main/.github/workflows/drift-detection.yml#L90

        id: drift
        run: |
          terramate run \
            --parallel 5 \
            --changed \

>             --sync-drift-status \

Reviewer: ```suggestion
            --sync-drift-status \
           --continue-on-error \
```
